File: stanfytest-rest/src/main/java/com/apischan/stanfytest/service/SkillService.java
package com.apischan.stanfytest.service;

import com.apischan.stanfytest.dto.SkillDto;

import java.util.List;

public interface SkillService {

    /**
     * Retrieve skill by it's instance id
     *
     * @param skillId instance id of skill
     * @return skill dto
     */
    SkillDto getSkillById(Integer skillId);

    /**
     * Retrieve skill by it's instance name
     *
     * @param skillName name of skill
     * @return skill dto
     */
    SkillDto getSkillByName(String skillName);

    /**
     * Retrieve all skill instances from database
     *
     * @return all skill dtos from database
     */
    List<SkillDto> getAllSkills();
}

File: stanfytest-akka-http-rest/src/main/scala/com/apischan/stanfytest/service/SkillService.scala
package com.apischan.stanfytest.service

import com.apischan.stanfytest.repository.SkillRepository
import com.google.inject.Injector

class SkillService(val skillRepository: SkillRepository) {

}

case class Skill(id: Int, name: String)

object SkillService {
  def apply()(implicit injector: Injector): SkillService =
    new SkillService(injector.getInstance(classOf[SkillRepository]))
}